src/kpagecount.rs

```diff
@@ -0,0 +1,76 @@
+use std::{fs::File, ops::Range, path::Path};
+
+use rustix::fs::FileExt;
+
+use super::ProcResult;
+
+/// Parse physical memory references accessing `/proc/kpagecount`
+///
+/// Require root or CAP_SYS_ADMIN
+pub struct KPageCount {
+    reader: File,
+}
+
+impl KPageCount {
+    /// Get a parser from default `/proc/kpagecount`
+    ///
+    /// Return `Err` if process is not running as root or don't have CAP_SYS_ADMIN
+    pub fn new() -> ProcResult<Self> {
+        Self::from_custom_root("/proc")
+    }
+
+    /// Get a parser from custom `/proc`
+    ///
+    /// Return `Err` if process is not running as root or don't have CAP_SYS_ADMIN
+    pub fn from_custom_root<P: AsRef<Path>>(root: P) -> ProcResult<Self> {
+        let mut path = root.as_ref().to_path_buf();
+        path.push("kpagecount");
+
+        let reader = File::open(&path)?;
```

Comment:
Can this use `FileWrapper::open(..)` instead?

Reply:
I don't know why I did it differently than kpageflags,

I refactored so the code for kpageflags and kpagecount is now similar.